transaction/core/src/mint/validation/tx.rs

```diff
@@ -0,0 +1,350 @@
+            validate_against_mint_config(&tx, &mint_config),
+            Err(Error::Signature)
+        );
+    }
+
+    #[test]
+    fn validate_signature_accepts_valid_signature() {
+        let mut rng: StdRng = SeedableRng::from_seed([1u8; 32]);
+        let token_id = 123;
+        let signer_1 = Ed25519Pair::from_random(&mut rng);
+
+        let prefix = MintTxPrefix {
+            token_id: token_id,
+            amount: 10,
+            view_public_key: RistrettoPublic::from_random(&mut rng),
+            spend_public_key: RistrettoPublic::from_random(&mut rng),
+            nonce: vec![1u8; NONCE_MIN_LENGTH],
+            tombstone_block: 10,
+        };
+        let message = prefix.hash();
+        let signature = MultiSig::new(vec![signer_1.try_sign(message.as_ref()).unwrap()]);
+        let tx = MintTx { prefix, signature };
+
+        let signer_set = SignerSet::new(vec![signer_1.public_key()], 1);
+
+        assert_eq!(validate_signature(&tx, &signer_set), Ok(()));
+    }
+
+    #[test]
+    fn validate_signature_rejects_signers_mismatch() {
+        let mut rng: StdRng = SeedableRng::from_seed([1u8; 32]);
+        let token_id = 123;
+        let signer_1 = Ed25519Pair::from_random(&mut rng);
+        let signer_2 = Ed25519Pair::from_random(&mut rng);
+
+        let prefix = MintTxPrefix {
+            token_id: token_id,
+            amount: 1,
+            view_public_key: RistrettoPublic::from_random(&mut rng),
+            spend_public_key: RistrettoPublic::from_random(&mut rng),
+            nonce: vec![1u8; NONCE_MIN_LENGTH],
+            tombstone_block: 10,
+        };
+        let message = prefix.hash();
+        let signature = MultiSig::new(vec![signer_1.try_sign(message.as_ref()).unwrap()]);
+        let tx = MintTx { prefix, signature };
+
+        let signer_set = SignerSet::new(vec![signer_2.public_key()], 1);
+
+        assert_eq!(validate_signature(&tx, &signer_set), Err(Error::Signature));
+    }
+
+    #[test]
+    fn validate_signature_rejects_tampered_messages() {
+        let mut rng: StdRng = SeedableRng::from_seed([1u8; 32]);
+        let token_id = 123;
+        let signer_1 = Ed25519Pair::from_random(&mut rng);
+
+        let prefix = MintTxPrefix {
+            token_id: token_id,
+            amount: 1,
+            view_public_key: RistrettoPublic::from_random(&mut rng),
+            spend_public_key: RistrettoPublic::from_random(&mut rng),
+            nonce: vec![1u8; NONCE_MIN_LENGTH],
+            tombstone_block: 10,
+        };
+        let message = prefix.hash();
+        let signature = MultiSig::new(vec![signer_1.try_sign(message.as_ref()).unwrap()]);
+        let signer_set = SignerSet::new(vec![signer_1.public_key()], 1);
+
+        let mut prefix1 = prefix.clone();
+        prefix1.amount = 2;
+        let tx = MintTx {
+            prefix: prefix1,
+            signature: signature.clone(),
+        };
+        assert_eq!(validate_signature(&tx, &signer_set), Err(Error::Signature));
```

Comment:
Optional: Also assert the base signer_set is valid

Reply:
Not sure what you mean by that.

Reply:
Assert the initial conditions are valid, as a sanity check before testing modified ones

e.g.
```
let tx = MintTx { prefix, signature: signature.clone() };
assert_eq!(validate_signature(&tx, &signer_set), Ok(()), "sanity check");
```